tests/Objects/GeometryCollectionTest.php

```diff
@@ -52,22 +52,41 @@
 });
 
 it('creates geometry collection from JSON', function (): void {
-  $geometryCollection = new GeometryCollection([
-    new Polygon([
-      new LineString([
-        new Point(0, 180),
-        new Point(1, 179),
-        new Point(2, 178),
-        new Point(3, 177),
-        new Point(0, 180),
+    $geometryCollection = new GeometryCollection([
+      new Polygon([
+        new LineString([
+          new Point(0, 180),
+          new Point(1, 179),
+          new Point(2, 178),
+          new Point(3, 177),
+          new Point(0, 180),
+        ]),
       ]),
-    ]),
-    new Point(0, 180),
-  ]);
+      new Point(0, 180),
+    ]);
 
-  $geometryCollectionFromJson = GeometryCollection::fromJson('{"type":"GeometryCollection","geometries":[{"type":"Polygon","coordinates":[[[180,0],[179,1],[178,2],[177,3],[180,0]]]},{"type":"Point","coordinates":[180,0]}]}');
+    $geometryCollectionFromJson = GeometryCollection::fromJson('{"type":"GeometryCollection","geometries":[{"type":"Polygon","coordinates":[[[180,0],[179,1],[178,2],[177,3],[180,0]]]},{"type":"Point","coordinates":[180,0]}]}');
 
-  expect($geometryCollectionFromJson)->toEqual($geometryCollection);
+    expect($geometryCollectionFromJson)->toEqual($geometryCollection);
+});
+
+it('creates geometry collection from array (serialized model)', function (): void {
+    $geometryCollection = new GeometryCollection([
+      new Polygon([
+        new LineString([
+          new Point(0, 180),
+          new Point(1, 179),
+          new Point(2, 178),
+          new Point(3, 177),
+          new Point(0, 180),
+        ]),
+      ]),
+      new Point(0, 180),
+    ]);
+
+    $geometryCollectionFromArray = GeometryCollection::fromArray($geometryCollection->toArray());
```

Comment:
I tend to keep the tests following the [AAA pattern](https://github.com/goldbergyoni/javascript-testing-best-practices#-%EF%B8%8F-12-structure-tests-by-the-aaa-pattern) - so anything that is related to arranging the data should be above the "act" part.
```suggestion
    $geometryCollection = new GeometryCollection([
      new Polygon([
        new LineString([
          new Point(0, 180),
          new Point(1, 179),
          new Point(2, 178),
          new Point(3, 177),
          new Point(0, 180),
        ]),
      ]),
      new Point(0, 180),
    ]);
    $geometryCollectionArray = $geometryCollection->toArray();

    $geometryCollectionFromArray = GeometryCollection::fromArray($geometryCollectionArray);
```